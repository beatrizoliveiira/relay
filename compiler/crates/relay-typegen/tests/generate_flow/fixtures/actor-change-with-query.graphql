fragment ActorFragment on User {
  name
  profilePicture {
    uri
    width
    height
  }
}

query Viewer {
  viewer {
    actor @EXPERIMENTAL__as_actor {
      ...ActorFragment
    }
  }
}
