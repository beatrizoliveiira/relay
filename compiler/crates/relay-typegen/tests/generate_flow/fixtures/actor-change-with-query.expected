==================================== INPUT ====================================
fragment ActorFragment on User {
  name
  profilePicture {
    uri
    width
    height
  }
}

query Viewer {
  viewer {
    actor @EXPERIMENTAL__as_actor {
      ...ActorFragment
    }
  }
}
==================================== OUTPUT ===================================
import type { ActorChangePoint } from "react-relay/multi-actor";
import type { ActorFragment$ref } from "ActorFragment.graphql";
export type ViewerVariables = {||};
export type ViewerResponse = {|
  +viewer: ?{|
    +actor: ?ActorChangePoint<{|
      +__viewer: string,
      +$fragmentRefs: ActorFragment$ref,
    |}>,
  |},
|};
export type Viewer = {|
  variables: ViewerVariables,
  response: ViewerResponse,
|};
-------------------------------------------------------------------------------
import type { FragmentReference } from "relay-runtime";
declare export opaque type ActorFragment$ref: FragmentReference;
declare export opaque type ActorFragment$fragmentType: ActorFragment$ref;
export type ActorFragment = {|
  +name: ?string,
  +profilePicture: ?{|
    +uri: ?string,
    +width: ?number,
    +height: ?number,
  |},
  +$refType: ActorFragment$ref,
|};
export type ActorFragment$data = ActorFragment;
export type ActorFragment$key = {
  +$data?: ActorFragment$data,
  +$fragmentRefs: ActorFragment$ref,
  ...
};
